### PR TITLE
LPS-145927

### PR DESCRIPTION
**Pull request by marianoalvarosaiz**

### CODE REVIEW

**Comment by marianoalvarosaiz** [comment]
Hi @catarineaguiar,
 I fixed all the unit tests you pointed. Is there someone else missing?
Kind regards.

**Comment by catarineaguiar** [comment]
hi @marianoalvarosaiz, everything looks OK now. No regressions were found.

**Comment by tinatian** [comment]
@dantewang Please take this over.

Thanks.
Tina.

File: portal-impl/src/com/liferay/portal/model/impl/UserImpl.java [hunk 1]
```diff
@@ -905,7 +905,12 @@ public void setDigest(String digest) {
 
 	@Override
 	public void setLanguageId(String languageId) {
-		_locale = LocaleUtil.fromLanguageId(languageId);
+		if (isDefaultUser()) {
+			_locale = LocaleUtil.fromLanguageId(languageId, false);
```
Comment (dantewang) [inline portal-impl/src/com/liferay/portal/model/impl/UserImpl.java:909]: @marianoalvarosaiz Could you point out which part in `LanguageImpl$ComapanyLocalesBag` constructor are you referring to?

Comment (marianoalvarosaiz) [reply portal-impl/src/com/liferay/portal/model/impl/UserImpl.java:909]: @dantewang https://github.com/liferay/liferay-portal/blob/master/portal-impl/src/com/liferay/portal/language/LanguageImpl.java#L2086-L2109

Comment (dantewang) [reply portal-impl/src/com/liferay/portal/model/impl/UserImpl.java:909]: @marianoalvarosaiz Thanks!

**Comment by dantewang** [comment]
@tinatian For https://github.com/liferay-core-infra/liferay-portal/pull/721/commits/36f5e0a3a856f74986a71b4716bfc97e76722945, with the code @marianoalvarosaiz marked, when the `if` block is executed, the default user's locale will be added to the `_languageIdLocalesMap` and the `_languageCodeLocalesMap` in the `CompanyLocalesBag`, even if the locale is not in the company's available locale list obtained using `PrefsPropsUtil`. The maps are then used in the `LanguageImpl` APIs like `isAvailableLocale`.

Based on the above logic, the default user's locale itself should not be validated against `isAvailableLocale`; instead, it should be always taken as a possible addition to the available locale list.

This change can also break the circle.

The PR looks good to me.

**Comment by vicnate5** [comment]
Triggered some additional testing for this
https://github.com/vicnate5/liferay-portal/pull/2038

**Comment by vicnate5** [comment]
No regressions found in further testing. @marianoalvarosaiz This does need a rebase though.

**Comment by marianoalvarosaiz** [comment]
@vicnate5 rebased: https://github.com/liferay-core-infra/liferay-portal/pull/743

**Pull request closed without merging**